src/types/message.rs

```diff
@@ -0,0 +1,29 @@
+//use crate::types::packet::Fragment;
+struct Message {}
+pub struct Fragment {}
+impl Message {
+    #[allow(unused_variables)]
+    fn serialize(&self) -> String {
+        unimplemented!()
+    }
```

Comment:
Having a struct with methods unimplemented does not seem right and this code. Because this code should be implemented by groups not class, we maybe should use a trait instead